apps/server/src/modules/oauth-provider/controller/oauth-provider.controller.ts

```diff
@@ -0,0 +1,115 @@
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Body, Controller, Delete, Get, NotImplementedException, Param, Patch, Post, Put, Query } from '@nestjs/common';
+import { Authenticate } from '@src/modules/authentication/decorator/auth.decorator';
+import { OauthProviderUc } from '@src/modules/oauth-provider/uc/oauth-provider.uc';
+import { OauthProviderResponseMapper } from '@src/modules/oauth-provider/mapper/oauth-provider-response.mapper';
+import {
+	AcceptQuery,
+	ChallengeParams,
+	ConsentRequestBody,
+	IdParams,
+	IntrospectBody,
+	ListOauthClientsParams,
+	LoginRequestBody,
+	OauthClientBody,
+	RedirectBody,
+	RevokeConsentQuery,
+	UserParams,
+} from './dto';
+
+@Controller('oauth2')
+export class OauthProviderController {
+	constructor(
+		private readonly oauthProviderUc: OauthProviderUc,
+		private readonly oauthProviderResponseMapper: OauthProviderResponseMapper
+	) {}
+
+	@Authenticate('jwt')
+	@Get('clients/:id')
+	getOAuth2Client(@Param() { id }: IdParams) {
```

Comment:
wie @SevenWaysDP richtig angemerkt hat, wollen wir es vermeiden in den Parametern zu destructuren. Wo erforderlich können die parameter innerhalb der funktion destructured werden.

hier also `getOAuth2Client(@Param() urlParams: IdParams` oder ähnliches.
Bitte an diversen stellen hier im PR kurz korrigieren :)